samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1184,17 +1316,39 @@ int main(void)
 			continue;
 		}
 
-		sync_bitfield = bis_index_bitfield & requested_bis_sync;
-		stream_count = 0;
-		for (int i = 0; i < BT_ISO_MAX_GROUP_ISO_COUNT; i++) {
-			if ((sync_bitfield & BIT(i)) != 0) {
-				stream_count++;
-			}
+		/* Only base BIS indexes from requested subgroups are of interest */
+		base_bis_index_bitfield &= get_subgroup_mask(&bis_index, requested_subgroup_sync);
+
+		/* If requested BIS indexes are 'no_pref' then adjust number of BIS indexes to
+		 * number of streams supported
+		 */
+		if (requested_bis_sync == BT_BAP_BIS_SYNC_NO_PREF) {
+#if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
+			/* Keep the n least significant bits of base_bis_index_bitfield to match the
+			 * target allocation
+			 */
+			base_bis_index_bitfield = keep_least_significant_allocation_match(
+				base_bis_index_bitfield, &bis_index.audio_allocation[0],
+				CONFIG_TARGET_BROADCAST_CHANNEL);
+
+#else  /* !CONFIG_TARGET_BROADCAST_CHANNEL */
+			/* Keep the CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT least significant bits
+			 * of base_bis_index_bitfield, as that is the maximum number of BISes we can
+			 * sync to
+			 */
+			base_bis_index_bitfield = keep_n_least_significant_ones(
+				base_bis_index_bitfield, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+#endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
 		}
 
+		printk("Adjusted base_bis_index_bitfield 0x%0x\n", base_bis_index_bitfield);
```

Comment:
```suggestion
		printk("Adjusted base_bis_index_bitfield 0x%08x\n", base_bis_index_bitfield);
```

To be consisted with previous prints of a 32-bit bitfield

Reply:
What is the purpose of this print? Adjust from what to what?

Reply:
Agree. Printout no longer needed. Added during debugging